### PR TITLE
pick_first: de-experiment pick first

### PR DESCRIPTION
**Pull request by mohanli-ml**
De-experiment pick first since we have both affinity and randomness E2E test running successfully.

@apolcyn @easwars 

RELEASE NOTES: none

### CODE REVIEW

**Review by dfawley — APPROVED** [review]

**Comment by easwars** [comment]
Looks like a test is failing. The failing test is this one: https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go#L185-L205

The test logic assumes that pick_first env var is disabled by default and enables it when the corresponding field in the test table is set: https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go#L300-L303. We need to invert this logic for the current test to pass. We also need another similar test wherein pick_first and round_robin are configured through custom LB policies, and when pick_first env var is enabled (which will be by default with this PR), the test should verify that the output contains pick_first config and not round_robin.

**Comment by mohanli-ml** [comment]
> Looks like a test is failing. The failing test is this one: https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go#L185-L205
> 
> The test logic assumes that pick_first env var is disabled by default and enables it when the corresponding field in the test table is set: https://github.com/grpc/grpc-go/blob/master/xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go#L300-L303. We need to invert this logic for the current test to pass. We also need another similar test wherein pick_first and round_robin are configured through custom LB policies, and when pick_first env var is enabled (which will be by default with this PR), the test should verify that the output contains pick_first config and not round_robin.

Done. Can you take another look? Thanks!

File: xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go [hunk 1]
```diff
@@ -203,6 +203,27 @@ func (s) TestConvertToServiceConfigSuccess(t *testing.T) {
...
+					},
+					{
+						TypedExtensionConfig: &v3corepb.TypedExtensionConfig{
+							TypedConfig: testutils.MarshalAny(&v3roundrobinpb.RoundRobin{}),
+						},
+					},
+				},
+			},
+			wantConfig: `[{"pick_first": { "shuffleAddressList": true }}]`,
+			pfDisabled: false,
```
Comment (easwars) [inline xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go:225]: Nit: we can omit this line as the default (zero value) for a boolean is `false`, unless you feel that explicitly including this line here improves the readability of the test.

Comment (mohanli-ml) [reply xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go]: Done. Thanks!

**Review by easwars — APPROVED** [review]

**Pull request merged**